### PR TITLE
[Synthetics] project fields for lightweight monitors

### PR DESCRIPTION
**Pull request by dominiqueclarke**
Adds project fields to lightweight project monitors via processors.

Screen Shot 2022-09-29 at 1 23 35 PM

### Testing
1. Pull down this PR
2. `cd packages/synthetics`
3. `elastic-package clean`
4. `elastic-package build`
5. `elastic-package stack up --version 8.5.0-SNAPSHOT -v`
6. Navigate to Uptime. Create a private location with the elastic-package agent policy
7. In your terminal, run `npx @elastic/synthetics init` to init a new synthetics testing directory, or update an existing testing directory to by updating `synthetics.config.ts` to use `https://localhost:5601` and update private locations to include your new private location. 
8. In your `journeys` directory within your synthetics tests, add the following lightweight monitors in a `heartbeat.yml` file
```
heartbeat.monitors:
- type: icmp
  id: stuff
  name: Cloudflare DNS
  hosts: ["1.1.1.1"]
  tags: ["service:dns", "org:cloudflare"]
- type: http
  name: A name
  id: http-monitor-with-host
  urls: ["http://www.doordash.com"] 
  tags: ["service:dns", "org:cloudflare"]
- type: tcp
  id: gmail-smtp
  name: GMail SMTP
  hosts: ["smtp.gmail.com:587"]
  tags: ["service:smtp", "org:google"]
```
10. Run `eval "$(elastic-package stack shellinit)" && export NODE_EXTRA_CA_CERTS=${ELASTIC_PACKAGE_CA_CERT}` within the synthetics directory
11. Navigate to Uptime Monitor Management. Generate an api key
12. In the synthetics directory, run `SYNTHETICS_API_KEY={THEAPIKEY} npm run push`
13. In Uptime, confirm that the project fields exist on the resulting documents, either in discover or dev tools

### CODE REVIEW

File: packages/synthetics/data_stream/http/agent/stream/http.yml.hbs [hunk 1]
```diff
@@ -80,3 +80,9 @@ processors:
         {{#if run_once}}
         run_once: {{run_once}}
         {{/if}}
+        {{#if monitor.project.name}}
```
Comment (dominiqueclarke) [inline packages/synthetics/data_stream/http/agent/stream/http.yml.hbs:83]: Confirmed that the mappings already exist for http, tcp and icmp data streams. FYI

**Review by shahzad31 — APPROVED** [review]

File: packages/synthetics/changelog.yml [hunk 2]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "0.10.3"
+  changes:
+    - description: Adds project fields for lightweight monitors
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/3998
```
Comment (dominiqueclarke) [inline packages/synthetics/changelog.yml:6]:
```suggestion
      link: https://github.com/elastic/integrations/pull/4326
```

**Pull request merged**